sourced/ml/core/algorithms/token_parser.py

```diff
@@ -166,6 +187,22 @@ def ret(name):
             if last:
                 yield from ret(last)
 
+    def split(self, token):
+        """
+        Splits a single identifier.
+        """
+        if self._use_nn:
+            return self._id_splitter_nn.split([token])
```

Comment:
`self._id_splitter_nn.split` - is it a generator? If it is not, we've got a return type inconsistency

Reply:
It is not. Shall I convert the output to one to guarantee return type consistency ?

Reply:
It should be consistent with the previous version, which is a generator

Reply:
resolved. 4692b5c